dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaOutgoingConnector.java

```diff
@@ -54,6 +54,9 @@
                                                           .filter(k -> !KafkaConnectorConstants.NON_KAFKA_PROPS.contains(k))
                                                           .collect(Collectors.toMap(Function.identity(), (k) -> config.getValue(k, String.class)));
 
+        producerConfig.putIfAbsent(KafkaConnectorConstants.KEY_SERIALIZER, KafkaConnectorConstants.DEFAULT_SERIALIZER);
```

Comment:
I don't like modifying the map returned from `collect()`. I don't think there's any guarantee in the API that it's mutable.

Reply:
I'd prefer it if this class used the same structure as `KafkaIncomingConnector` for handling config defaults.